sample/sample8-middlewares/UserController.ts

```diff
@@ -0,0 +1,56 @@
+import {Request} from "express";
+import {Controller} from "../../src/decorator/controllers";
+import {Get, Post, Put, Patch, Delete, Use} from "../../src/decorator/methods";
+import {Req, Next} from "../../src/decorator/params";
+
+@Controller()
+export class UserController {
+
+
+    @Use()
+    middleware( @Next() next: any) {
+      console.log("calling middleware");
+      next();
+    }
+
+    @Get("/users", { jsonResponse: true })
+    getAll() {
+        return [
+            { id: 1, name: "First user!" },
+            { id: 2, name: "Second user!" }
+        ];
+    }
+
+    @Get("/users/:id")
+    getOneMdlw( @Req() request: Request, @Next() next: any) {
```

Comment:
I used Next in case we need middleware for a specific HTTP verb.

In this case we use the method decorator (Get, Post,...) and the Next function for passing to the next route...